### ISSUE TITLE
Confusion with geopandas dataframe as Polygons input

### ISSUE THREAD

**Issue by msarahan** [pyviz/geoviews#214]
I have a GDF that I've created by overlaying a few other GDF's (state district data for various offices).  I want to plot that on top of tile data.

Plotting the GDF with matplotlib via geopandas seems to work OK:

![travis](https://user-images.githubusercontent.com/38393/44004731-bc83a330-9e2c-11e8-99fe-842045e5ed8e.png)

but creating a Polygons object seems to go wrong:

<img width="1268" alt="screen shot 2018-08-12 at 12 54 45" src="https://user-images.githubusercontent.com/38393/44004830-fff7feb6-9e2e-11e8-9c3d-73467042781a.png">

My ultimate goal is a colorized map with a hover tool: each district's color represents a unique combination of districts that cover that area, and the hover tool will tell you which districts cover a given area.  In Travis County, there are 58 such combinations, so a legend is not really feasible.  Color-coding would still be nice, though, to make the separations more clear.

I'm happy to supply the GDF - please suggest the best way to do so.

**Comment by philippjfr** [comment]
That looks pretty bad. One issue that may be causing this is that bokeh does not yet have support for polygon holes, so if there is any negative space inside the polygons you're going to get overlap. We are currently working on a project that explicitly requires this support, so I'm hoping to start working on that soon. 

>I'm happy to supply the GDF - please suggest the best way to do so.

Just in case it's not that, could you email it to me?

**Comment by rsignell-usgs** [comment]
https://github.com/bokeh/bokeh/pull/8340

**Comment by philippjfr** [comment]
Thanks @rsignell-usgs, that PR was actually sponsored by our ongoing project and I was going to start implementing this in holoviews/geoviews today.

**Comment by philippjfr** [comment]
@msarahan I am encountering some weirdness which is causing various issues when projecting the data, e.g. this polygon seems very strange and when projected ends up hiding other polygons behind it:

![screen shot 2018-10-21 at 5 36 39 pm](https://user-images.githubusercontent.com/1550771/47269578-f8552200-d557-11e8-9ec2-5d0cd5debac1.png)

**Comment by philippjfr** [comment]
The exact issue can be reproduced with:

```
wkt = 'POLYGON ((-97.6453334387044833 30.4773161392542349, -97.6453334387044691 30.4773161392542455, -97.6453334387044549 30.4773161392542491, -97.6448419999999970 30.4775050000000007, -97.6453334387044833 30.4773161392542349))'
poly = shapely.wkt.loads(wkt)
poly
```

![screen shot 2018-10-21 at 5 54 32 pm](https://user-images.githubusercontent.com/1550771/47269771-67337a80-d55a-11e8-9142-a739e30784b9.png)

```python
ccrs.GOOGLE_MERCATOR.project_geometry(poly, ccrs.PlateCarree())
```

![screen shot 2018-10-21 at 5 55 02 pm](https://user-images.githubusercontent.com/1550771/47269778-803c2b80-d55a-11e8-9287-f49dd917717b.png)

**Comment by philippjfr** [comment]
Dropping all polygons with an area < 1e-15 seems to fix things, suspect it's a floating point precision issue in the GEOS projection code.

**Comment by philippjfr** [comment]
Once merged this will work:

```python
%%opts Polygons [width=800 height=600 tools=['hover', 'tap'] show_bounds=False] (cmap='gist_ncar' alpha=0.5)
intersection = intersection.reset_index(drop=True)
intersection_p = gv.Polygons(intersection, vdims=['index', 'Congress_CD115FP', 'State_house_NAME', 'State_senate_NAME']).options(alpha=0.5)
gv.tile_sources.Wikipedia * intersection_p
```

![screen shot 2018-10-21 at 6 08 40 pm](https://user-images.githubusercontent.com/1550771/47269943-5edc3f00-d55c-11e8-8cd4-e27e97e4fb09.png)

**Comment by msarahan** [comment]
Indeed, by plotting each geometry, it was clear by inspection that some were just tiny slivers where the boundaries didn't exactly match up.  Floating point error is a great explanation.  Hooray for crummy data.

**Comment by jbednar** [comment]
Should we have an optional utility that someone could call to match up slightly mismatched boundaries and eliminate tiny slivers, using a heuristic based on the total area enclosed (to determine how big a "sliver" is)?

**Comment by philippjfr** [comment]
Once my PR is merged we will automatically drop slivers with an area <1e-15, which seemingly cannot be accurately projected by GEOS.

**Comment by philippjfr** [comment]
>Should we have an optional utility that someone could call to match up slightly mismatched boundaries 

Also, this kind of thing is definitely out-of-scope for geoviews. It's something that should be handled at the shapely or even GEOS level.

**Comment by jbednar** [comment]
I've never gotten very far with opinions about what *should* be handled by some lower-level library I don't control. :-)  In practice if it happens for users at a higher level, it has to be dealt with *somehow*, which usually means hacks at the very highest level (user scripts)...

**Comment by philippjfr** [comment]
>In practice if it happens for users at a higher level, it has to be dealt with somehow, which usually means hacks at the very highest level (user scripts)...

These slivers are generated by applying intersect/overlap operations on shapely objects, which is not the responsibility of geoviews to deal with.
